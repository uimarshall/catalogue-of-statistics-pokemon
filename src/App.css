.App {
  display: grid;
  grid-gap: 20px;
}
