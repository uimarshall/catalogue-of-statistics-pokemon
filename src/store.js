/* eslint-disable no-underscore-dangle */
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const initialState = {
  pending: true,
  pendingPokemon: true,
  pokemons: [],
  error: null,
};

const middleware = [thunk];
const devtools = process.env.NODE_ENV === 'test'
  ? x => x
  : window.__REDUX_DEVTOOLS_EXTENSION__
      && window.__REDUX_DEVTOOLS_EXTENSION__();

const store = createStore(
  rootReducer,
  {
    data: initialState,
    filter: 'normal',
  },
  compose(
    applyMiddleware(...middleware),
    devtools,
  ),
);

export default store;
