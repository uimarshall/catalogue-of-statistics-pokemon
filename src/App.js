import { lazy, Suspense } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import {
  HOME, POKEMON_DETAILS, NOT_FOUND, SINGLE_POKEMON,
} from './constants/routes';
import './App.css';
import Spinner from './utils/Spinner';

const Home = lazy(() => import('./containers/PokemonItems'));
const Card = lazy(() => import('./containers/Card'));
const PokemonDetails = lazy(() => import('./components/PokemonDetails'));
const Error404Page = lazy(() => import('./components/Error404Page'));

function App() {
  return (
    <div className="App">
      <Router>
        <Switch>
          <Route exact path={HOME}>
            <Suspense fallback={Spinner()}>
              <Home />
            </Suspense>
          </Route>
          <Route path={POKEMON_DETAILS}>
            <Suspense fallback={Spinner()}>
              <PokemonDetails />
            </Suspense>
          </Route>
          <Route path={SINGLE_POKEMON}>
            <Suspense fallback={Spinner()}>
              <Card />
            </Suspense>
          </Route>
          <Route path={NOT_FOUND}>
            <Suspense fallback={Spinner()}>
              <Error404Page />
            </Suspense>
          </Route>

        </Switch>
      </Router>

    </div>
  );
}

export default App;
