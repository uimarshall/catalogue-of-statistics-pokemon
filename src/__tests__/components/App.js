/* eslint-disable import/no-duplicates */
import '@testing-library/jest-dom';

import * as React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../../App';
import Link from '../../App';
import store from '../../store';

describe('App', () => {
  test('renders App component', () => {
    const nonExist = 'No such text';

    render(<Provider store={store}><App /></Provider>);

    expect(screen.queryByText(nonExist)).toBeNull();
  });

  afterEach(cleanup);

  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Link to="/">Home</Link></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Link to="/details">PokemonDetails</Link></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Link to="*">Error404Page</Link></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
