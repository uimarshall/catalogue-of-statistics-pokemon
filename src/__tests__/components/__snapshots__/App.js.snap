// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="d-flex justify-content-center align-items-center spinner"
  >
    <div
      className="spinner-inner spinner-border text-warning"
    />
  </div>
</div>
`;

exports[`App renders correctly 2`] = `
<div
  className="App"
>
  <div
    className="d-flex justify-content-center align-items-center spinner"
  >
    <div
      className="spinner-inner spinner-border text-warning"
    />
  </div>
</div>
`;

exports[`App renders correctly 3`] = `
<div
  className="App"
>
  <div
    className="d-flex justify-content-center align-items-center spinner"
  >
    <div
      className="spinner-inner spinner-border text-warning"
    />
  </div>
</div>
`;
